pkgs/development/python-modules/setuptools/default.nix

```
@@ -27,6 +28,14 @@ let
     patches = [
       ./tag-date.patch
       ./setuptools-distutils-C++.patch
+      # Fix cross compilation of extension modules
+      # https://github.com/pypa/distutils/pull/173
+      (fetchpatch {
+        url = "https://github.com/pypa/distutils/commit/22b9bcf.patch";
```

```suggestion
        url = "https://github.com/pypa/distutils/commit/22b9bcf2e2d2a66f7dc96661312972e2f6bd9e01.patch";
```
using short commits is forbidden because their are dos'able

---

Uhm, I guess if someone pushed 20 million commits to distutils, maybe.
Who says it's forbidden, btw? There's nothing in the contributor's guide or the Nixpkgs manual.

---

> Uhm, I guess if someone pushed 20 million commits to distutils, maybe.

Since github is sharing commits between forks, it is enough to push a commit with the same prefix into a fork which causes github to serve a 404. This is a practical attack https://blog.teddykatz.com/2019/11/12/github-actions-dos.html . and nixpkgs already reached the point where some 6 character long commits are longer served https://github.com/NixOS/nixpkgs/commit/c9d59b or https://github.com/NixOS/nixpkgs/commit/c9d59c but not https://github.com/NixOS/nixpkgs/commit/c9d59d



> Who says it's forbidden, btw? There's nothing in the contributor's guide or the Nixpkgs manual.

I already once converted most short sha to long shas which revealed some ambiguous ones.

---

Ok, so please document this in the manual instead of keeping it as Nixpkgs lore.

---

Do you have an idea which part of the doc fits this?

---

Since this is specifically a GitHub issue, I'd say in https://nixos.org/manual/nixpkgs/stable/#fetchfromgithub

---

https://github.com/NixOS/nixpkgs/pull/189241